packages/grafana-ui/src/components/PanelChrome/PanelChrome.tsx

```diff
@@ -1,20 +1,43 @@
-import { css } from '@emotion/css';
+import { css, cx } from '@emotion/css';
 import React, { CSSProperties, ReactNode } from 'react';
 
-import { GrafanaTheme2 } from '@grafana/data';
+import { GrafanaTheme2, LoadingState } from '@grafana/data';
 
 import { useStyles2, useTheme2 } from '../../themes';
+import { IconName } from '../../types/icon';
+import { IconButton } from '../IconButton/IconButton';
+import { PopoverContent } from '../Tooltip';
+
+/**
+ * @internal
+ */
+interface PanelChromeInfoState {
+  icon: IconName;
+  tooltip?: PopoverContent;
+  onClick?: () => void;
+}
 
 /**
  * @internal
  */
 export interface PanelChromeProps {
   width: number;
   height: number;
-  title?: string;
+  children: (innerWidth: number, innerHeight: number) => ReactNode;
   padding?: PanelPadding;
-  leftItems?: React.ReactNode[]; // rightItems will be added later (actions links etc.)
-  children: (innerWidth: number, innerHeight: number) => React.ReactNode;
+  title?: string;
+  titleItems?: PanelChromeInfoState[];
+  menu?: ReactNode;
+  dragClass?: string;
```

Comment:
When it comes to the ability of dragging a panel, I don't think the PanelChrome should have knowledge about the `dragClass` - as the dragging mechanism can change in the future and not rely on the class name, and the property would not be needed anymore.

I think we have two options here:
1. Give full control of the drag handle to the consumer of PanelChrome - let him/her provide the drag handle via `titleItems` array - looking at the design that's exactly what the drag handle is.
2. Instead of `dragClass` have a property `dragHandle: ReactNode` that explicitly requires a react element to be passed by the consumer.

IMO the second option is the most optimal and makes the PanelChrome possible to use with any kind of dnd mechanism.

Reply:
I'm not yet quite on board with how the drag-n-drop works exactly, so I cannot comment for now which one of those is better imo. Thank you for starting the conversation though - I've added your comments to the [Epic](https://github.com/grafana/grafana/issues/56163) under _To Be Discussed_ in order to tackle it on the next iteration of changes here.